src/dsp_tools/utils/id_to_iri.py

```diff
@@ -207,5 +239,10 @@ def id_to_iri(
         tree=tree,
```

Comment:
Is it on purpose that success can be overwritten by the method '_remove_resources_if_id_in_mapping' which follows?